doc/exception_interrupts.rst

```diff
@@ -18,12 +18,43 @@ It is assumed that the boot address is supplied via a register to avoid long pat
 Interrupts
 ----------
 
-Interrupts can only be enabled/disabled on a global basis and not individually.
-The global interrupt enable is done via the ``mstatus`` CSR.
+In Ibex, interrupts are controlled via the ``mstatus``, ``mie`` and ``mip`` CSRs.
+The following interrupts are supported.
 
-It is assumed that there is a separate event/interrupt controller outside of the core that performs masking and buffering of multiple interrupt requests.
-When an interrupt is taken, the core gives an acknowledge signal to the external event/interrupt controller as well as the interrupt ID taken.
-Check :ref:`interrupts` for more details.
++-------------------------+-------+--------------------------------------------------+
+| Interrupt Input Signal  | ID    | Description                                      |
++=========================+=======+==================================================+
+| ``irq_fast_i[14:0]``    | 30:16 | 15 fast, local interrupts                        |
++-------------------------+-------+--------------------------------------------------+
+| ``irq_external_i``      | 11    | Connected to platform-level interrupt controller |
++-------------------------+-------+--------------------------------------------------+
+| ``irq_timer_i``         | 7     | Connected to timer module                        |
++-------------------------+-------+--------------------------------------------------+
+| ``irq_software_i``      | 3     | Connected to memory-mapped (inter-processor)     |
+|                         |       | interrupt register                               |
++-------------------------+-------+--------------------------------------------------+
+
+All interrupt lines are level-sensitive.
+It is assumed that the interrupt handler signals completion of the handling routine to the interrupt source, e.g., through a memory-mapped register, which then deasserts the corresponding interrupt line.
+
+If multiple interrupts are pending, the highest priority is given to the interrupt with the highest ID.
+
+After reset, all interrupts are disabled.
+To enable interrupts, both the global interrupt enable (MIE) bit in the ``mstatus`` CSR and the corresponding individual interrupt enable bit in the ``mie`` CSR need to be set.
+For more information, see the :ref:`cs-registers` documentation.
+
+In debug mode, interrupts are ignored independent of ``mstatus``.MIE and the content of the ``mie`` CSR.
+
+
+Non-Maskeable Interrupt
+-----------------------
```

Comment:
A NMI is also an interrupt, and I'd prefer it being listed as part of the previous section "Interrupts".

Reply:
Yes. I moved the NMI to the interrupt section. But I have kept a separate section for the recoverable NMI stuff as this is non-stanard.